src/blocks.zig

```diff
@@ -414,39 +418,108 @@ pub const Block = packed struct { // MARK: Block
 };
 
 // MARK: Tick
-pub var tickFunctions: utils.NamedCallbacks(TickFunctions, TickFunction) = undefined;
-pub const TickFunction = fn(block: Block, _chunk: *chunk.ServerChunk, x: i32, y: i32, z: i32) void;
-pub const TickFunctions = struct {
-	pub fn replaceWithCobble(block: Block, _chunk: *chunk.ServerChunk, x: i32, y: i32, z: i32) void {
-		std.log.debug("Replace with cobblestone at ({d},{d},{d})", .{x, y, z});
-		const cobblestone = parseBlock("cubyz:cobblestone");
+pub const TickEventVTableMap = struct {
+	const Self = @This();
 
-		const wx = _chunk.super.pos.wx + x;
-		const wy = _chunk.super.pos.wy + y;
-		const wz = _chunk.super.pos.wz + z;
+	vTableMap: std.StringHashMap(*const TickEvent.VTable) = undefined,
+
+	pub fn init() Self {
+		var self = Self{
+			.vTableMap = .init(main.globalAllocator.allocator),
+		};
+
+		switch(@typeInfo(TickEvents)) {
+			.@"struct" => |eventStruct| {
+				inline for(eventStruct.decls) |decl| {
+					const field = @field(TickEvents, decl.name);
+					if(@typeInfo(@TypeOf(field)) == .type) {
+						registerEventStruct(field, &self);
+					}
+				}
+			},
+			else => {
+				std.log.err("Type {s} is not a struct", .{@typeName(TickEvents)});
+			},
+		}
+
+		return self;
+	}
+
+	pub fn deinit(self: *Self) void {
+		self.vTableMap.deinit();
+	}
+
+	fn registerEventStruct(comptime EventType: type, self: *Self) void {
+		inline for(@typeInfo(EventType).@"struct".decls) |eventDecl| {
+			const field = @field(EventType, eventDecl.name);
```

Comment:
Ideally you'd automatically construct the VTable here, instead of doing it manually on every tick event.